nchoosek/solver/common.py

```diff
@@ -58,7 +59,54 @@ def __init__(self):
         self.solutions = None
         self.tallies = None
         self.energies = None
-        self.jobs = None
         self.jobIDs = None
         self.qubits = None
         self.depth = None
+        self.times = None
+        self.quantum_instance = None
+
+    def __repr__(self):
+        ret = ""
+        if self.solutions:
+           ret += "top solution: " + str(self.solutions[0]) + "\n"
+           ret += "number of solutions: " + str(len(self.solutions)) + "\n"
+        if self.tallies:
+            ret += "top solution tallies: " + str(self.tallies[0]) + "\n"
+        if self.energies:
+            ret += "top solution energy: " + str(self.energies[0]) + "\n"
+        if self.qubits:
+            ret += "qubits: " + str(self.qubits) + "\n"
+        if self.depth:
+            ret += "depth: " + str(self.depth) + "\n"
+        if self.times:
+            ret += "times: " + str(self.times) + "\n"
+        if self.quantum_instance:
+            ret += "Qiskit backend: " + str(self.quantum_instance.backend) + "\n"
+        ret = ret[:-1]
+        return ret
```

Comment:
Let's have this function format the object as a single-line, syntactically valid Python dictionary.  This would be convenient if the user wanted to write result objects to a log file structured like "<timestamp>: <message>", for example.